trace.go

```diff
@@ -72,6 +72,16 @@ func Unwrap(err error) error {
 	return err
 }
 
+// ClearTraces empties the list of traces
+func ClearTraces(err error) error {
```

Comment:
Do we need this in the trace package? So far, this is special-case functionality that works on exported attributes which can be implemented where it is required.
Another question is whether the error types trace provides should be marshaled as they're. Can we consider a way of changing that both ways - e.g. revisit the http marshalling layer that outputs these values in the expected format but without the traces since they don't have any?